charms/slurmd/src/slurmd_ops.py

```diff
@@ -129,6 +129,47 @@ def install(self) -> bool:
 
         return package_installed
 
+class ROCmPackagesLifecycleManager:
+    """Facilitate ROCm packages lifecycles."""
+
+    _KEYRING_PATH = Path(f"/usr/share/keyrings/rocm.gpg")
+
+    def install(self) -> bool:
+        """Install ROCm packages using apt."""
+
+        if self._KEYRING_PATH.exists():
+            self._KEYRING_PATH.unlink()
+        self._KEYRING_PATH.write_text(ROCM_PPA_KEY)
+
+        amdgpu_repo = apt.DebianRepository(
+            enabled=True,
+            repotype="deb",
+            uri="https://repo.radeon.com/amdgpu/6.2/ubuntu",
+            release=distro.codename(),
+            groups=["main"],
+            gpg_key_filename=self._KEYRING_PATH,
+            options={
+                # TODO: See if there's a way to get the charm architecture from ops.
+                "arch": "amd64"
+            }
+        )
+
+        rocm_repo = apt.DebianRepository(
+            enabled=True,
+            repotype="deb",
+            uri="https://repo.radeon.com/rocm/apt/6.2",
+            release=distro.codename(),
+            groups=["main"],
+            gpg_key_filename=self._KEYRING_PATH,
+            options={
+                "arch": "amd64"
```

Comment:
I know this is draft, but please make sure we're not hardcoding this arch string in the future. There are other architectures in the mix (e.g. amd64, riscv64, etc).

Reply:
Definitely! I had a todo comment here on a previous version, but your review will be a good reminder to fix this.